src/Model/Behavior/SearchBehavior.php

```diff
@@ -101,7 +119,11 @@ public function searchManager()
      */
     protected function _extractParams($params, $filters)
     {
-        return array_intersect_key(Hash::filter($params), $filters);
+        return array_intersect_key(Hash::filter($params, function ($val) {
+            $emptyValues = (array)$this->getConfig('emptyValues');
```

Comment:
Move this statement out of the closure and pass `emptyValues` as  `function ($val) use ($emptyValues) {` to avoid making the same method call on each iteration.